### PR TITLE
(PCP-592) Add acceptance test for syntax significant characters

### PR DESCRIPTION
**Pull request by james-stocks**
Add an additional acceptance test that ensures that syntax significant characters are properly handled by PXP

[skip ci]

### CODE REVIEW

**Comment by james-stocks** [comment]
@MikaelSmith @lucywyman @zreichert As well as reviewing the code added here, we should review if the test coverage being added is worth the extra CI expense

**Comment by james-stocks** [comment]
Also note this largely copies an existing test; and refactoring the duplicate code should be done at some point

File: acceptance/tests/pxp-module-puppet/run_puppet_syntax_chars.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  env_name = test_file_name = File.basename(__FILE__, '.*')
+  environment_name = mk_tmp_environment(env_name)
+
+  step 'On master, create a new environment that includes syntax significant characters in resources and will result in Changed' do
+    site_manifest = "#{environmentpath}/#{environment_name}/manifests/site.pp"
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  notify {'\\'\\"&+/\\()?,.#!@$_-~':}
```
Comment (mruzicka) [inline acceptance/tests/pxp-module-puppet/run_puppet_syntax_chars.rb]: Would it make sense to include some characters outside of ASCII? ☃? Or is this covered by some other test?

Comment (james-stocks) [inline acceptance/tests/pxp-module-puppet/run_puppet_syntax_chars.rb]: pxp-module-puppet/run_puppet_nonASCII.rb already does that.

We _could_ throw all the characters into a single test; but our general testing strategy is to get a detailed test result set, so we want separate test results for multibyte characters and for potential poorly-escaped syntax characters

Comment (MikaelSmith) [inline acceptance/tests/pxp-module-puppet/run_puppet_syntax_chars.rb]: This seems like it mainly ensures pxp-module-puppet handles output safely. Could this be accomplished with a spec test instead?

Comment (james-stocks) [inline acceptance/tests/pxp-module-puppet/run_puppet_syntax_chars.rb]: I think you're right @MikaelSmith - this shouldn't be an acceptance test unless we suspect different behaviours on different platforms. If we trust the Ruby execution of pxp-module-puppet to be consistent across all agent platforms; this can be a spec test.

**Comment by james-stocks** [comment]
Closing this PR - the only thing being tested (that is the responsibility of this repo's code) is the parsing of Puppet stdout by pxp-module-puppet - that can be tested a lot quicker with a spec test
